HopsanGUI/GUIObjects/GUIContainerObject.cpp

```diff
@@ -4889,6 +4889,35 @@ void SystemObject::loadParameterValuesFromFile(QString parameterFile)
     emit checkMessages();
 }
 
+void SystemObject::getAllParametersAndValuesRecursively(QString prefix, QStringList &names, QStringList &values, QStringList &dataTypes, QStringList &quantities, QStringList &units)
+{
+    QVector<CoreParameterData> parameters;
+    getParameters(parameters);
+    for(const CoreParameterData &parameter : parameters) {
+        names.append(prefix+parameter.mName);
+        values.append(parameter.mValue);
+        dataTypes.append(parameter.mType);
+        quantities.append(parameter.mQuantity);
+        units.append(parameter.mUnit);
+    }
+
+    for(ModelObject *modelObject : getModelObjects()) {
+        if(modelObject->getTypeName() == "Subsystem") {
+            qobject_cast<SystemObject*>(modelObject)->getAllParametersAndValuesRecursively(prefix+modelObject->getName()+"|", names, values, dataTypes, quantities, units);
+            continue;
```

Comment:
Not a big deal, but I would have preferred a good old if ... else,  instead of continue to skip the "else part"

Reply:
You are right. I like to use "continue" in this way sometimes to avoid indenting very long sections of code, but in this case there is no point.